wrapper/src/main/java/software/amazon/jdbc/plugin/efm2/MonitorConnectionContext.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.efm2;
+
+import java.lang.ref.WeakReference;
+import java.sql.Connection;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Monitoring context for each connection. This contains each connection's criteria for whether a
+ * server should be considered unhealthy. The context is shared between the main thread and the monitor thread.
+ */
+public class MonitorConnectionContext {
+
+  private final AtomicReference<WeakReference<Connection>> connectionToAbortRef;
```

Comment:
Thoughts on simplifying this to a volatile?
```suggestion
  `private volatile WeakReference<Connection> connectionToAbortRef;
```

https://stackoverflow.com/questions/281132/java-volatile-reference-vs-atomicreference?rq=4

Reply:
Volatile doesn't prevent multi-threading update collisions. It just makes a note to compiler to not optimize the final byte code and to expect the variable value may change at any time.